sound/soc/sof/ipc4-topology.c

```diff
@@ -1632,7 +1634,27 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 	ipc_config_data = &ipc4_copier->ipc_config_data;
 
 	/* config_length is DWORD based */
-	ipc_size = sizeof(*copier_data) + copier_data->gtw_cfg.config_length * 4;
+	gtw_cfg_config_length = copier_data->gtw_cfg.config_length * 4;
+	ipc_size = sizeof(*copier_data) + gtw_cfg_config_length;
+
+	if (ipc4_copier->dma_config_tlv.type == SOF_IPC4_GTW_DMA_CONFIG_ID &&
+	    ipc4_copier->dma_config_tlv.length) {
```

Comment:
the `ipc4_copier->dma_config_tlv.length` is always at least `sizeof(struct sof_ipc4_dma_config))`, no? Or the comment in the structure needs to be updated to allow 0 for this when the dma_config is unused and length is 0?